docs/src/proposals/epoch_accounts_hash.md

```diff
@@ -0,0 +1,203 @@
+     bank hash.  Since the bank hash is part of what is voted on for consensus,
+     this means every account is verified by the network at least once per
+     epoch.
+
+However, there is a big downside to rewriting unchanged accounts: performance.
+Storing accounts can be very expensive.  And since accounts now are required to
+be rent-exempt, the majority of accounts rewritten due to rent collection are
+unchanged.  What if unchanged accounts were no longer rewritten? This would
+minimally be a big performance win.
+
+
+## Problem
+
+If rent collection no longer is rewriting unchanged accounts, we lose the two
+positive effects.  Dealing with Positive Effect 1 (from above) will be handled
+by _Ancient AppendVecs_, and will not be discussed here.  So how do we still
+get the security from Positive Effect 2?  How can we still verify every account
+at least once per epoch, *as part of consensus*, but without rewriting
+accounts?
+
+
+## Proposed Solution
+
+Perform a full accounts hash calculation once per epoch, and hash the result
+into a bank's `hash`.  This will be known as the _Epoch Accounts Hash_, or
+_EAH_.
+
+This retains the Positive Effect 2 from rent collection by checking every
+account at least once per epoch.  Thus, any validators with missing, corrupt,
+or extra accounts will identify those issues within 1-2 epochs.
+
+
+### Implementation
+
+Performing a full accounts hash takes a relatively long time to complete.  For
+this reason, the EAH calculation must take place in the background.
+
+In order for all the validators to calculate the same accounts hash, the
+calculation must be based on the same view of all accounts.  This means the EAH
+must be based on a predetermined slot.  This will be known as the `start slot`.
+The `start slot` is calculated as an offset from the first slot of an epoch.
+This offset will be known as the `start offset`. 
+
+Similarly, all the validators must save the EAH into a bank at a predetermined
+slot, and offset from the first slot of an epoch.  This will be known as the
+`stop slot` and `stop offset`, respectively.
+
+* The `start offset` will be set at one-quarter into the epoch.
+* The `stop offset` will be set at three-quarters into the epoch.
+* For epochs with 432,000 slots, the `start offset` will be 108,00 and the
+  `stop offset` will be 324,000.
+
+These constants may be changed in the future, or may be determined at runtime.
+The main justifications for these values are:
+1. Do not start the EAH calculation at the beginning of an epoch, as the
+   beginning of an epoch is already a time of contention and stress.  There is
+   no reason to make this worse.
+2. The bank to save the EAH into—the `stop offset`—should be sufficiently far
+   in the future to guarantee all validators are able to complete the accounts
+   hash calculation in time.
+3. The EAH window, `start offset` to `stop offset`, should be small.  The
```

Comment:
this window being small is not important for performance. The validator only has to keep around the account data for `start offset` long enough for the EAH to be calculated. Then, it can clean and shrink as normal.